huffman/test/bit_span_test.cpp

```diff
@@ -92,4 +92,72 @@ auto main() -> int
       huffman::bit_span{data.begin(), bit_size, bit_offset};
     }));
   } | std::vector<std::uint8_t>{8, 9, 10};  // NOLINT(readability-magic-numbers)
+
+  test("consume") = [] {
+    // NOLINTBEGIN(readability-magic-numbers)
+    static constexpr std::array data{
+        std::byte{0b10101010}, std::byte{0b01010101}};
+    huffman::bit_span span{data.data(), data.size() * CHAR_BIT};
+    expect(*span.begin() == 0_b);
+    expect(span.data() == data.data());
+    // should be a no-op now.
+    span.consume_to_byte_boundary();
+    expect(*span.begin() == 0_b);
+    expect(span.data() == data.data());
+
+    span.consume(1);
+    expect(*span.begin() == 1_b);
+    expect(span.data() == data.data());
+
+    span.consume(1);
+    expect(*span.begin() == 0_b);
+    expect(span.data() == data.data());
```

Comment:
two notes:

1. I think having a lot of mutation in a test makes it slower and/or harder to understand due to the amount of history I need to track. I think this would be easier to read with a value parameterized test such as:

```cpp
   test("consume") = [](auto n) {
     static constexpr std::array data{
         std::byte{0b10101010}, std::byte{0b01010101}};

     static constexpr auto nth_bit = [](auto m) {
       const auto i = m < 8U;
       return huffman::bit{
           std::bitset<CHAR_BIT>{std::to_integer<int>(data[i])}[m]};
     };

     auto span = huffman::bit_span{data};
     span.consume(n);

     expect(nth_bit(n) == span[0]);
     expect(CHAR_BIT * data.size() - n == span.size());
   } | std::views::iota(0U, 16U);
```

2. I don't think we should have `data()` return a pointer to a byte if there is a non-zero bit offset. `data()` should always return the first element of the range. If not, the semantics of this type differ slightly from the semantics of the standard library which can cause all sort of headaches down the line.

https://en.cppreference.com/w/cpp/ranges/data